xwrf/grid.py

```diff
@@ -0,0 +1,139 @@
+"""Provide grid-related functionality specific to WRF datasets.
+
+This submodule contains code reused with modification from Salem under the terms of the BSD
+3-Clause License. Salem is Copyright (c) 2014-2021, Fabien Maussion and Salem Development Team All
+rights reserved.
+"""
+
+import numpy as np
+import pyproj
+
+# Default CRS (lon/lat on WGS84, which is EPSG:4326)
+wgs84 = pyproj.CRS(4326)
+
+# WRF dimension names
+_HORIZONTAL_DIMS = ['south_north', 'west_east', 'south_north_stag', 'west_east_stag']
+
+
+def _wrf_grid_from_dataset(ds):
+    """Get the WRF projection and dimension coordinates out of the file."""
+
+    pargs = {'x_0': 0, 'y_0': 0, 'a': 6370000, 'b': 6370000}
+    if hasattr(ds, 'PROJ_ENVI_STRING'):
+        # HAR and other TU Berlin files
+        dx = ds.GRID_DX
+        dy = ds.GRID_DY
+        pargs['lat_1'] = ds.PROJ_STANDARD_PAR1
+        pargs['lat_2'] = ds.PROJ_STANDARD_PAR2
+        pargs['lat_0'] = ds.PROJ_CENTRAL_LAT
+        pargs['lon_0'] = ds.PROJ_CENTRAL_LON
+        pargs['center_lon'] = ds.PROJ_CENTRAL_LON
+        if ds.PROJ_NAME in ['Lambert Conformal Conic', 'WRF Lambert Conformal']:
+            proj_id = 1
+        else:
+            proj_id = 99  # pragma: no cover
+    else:
+        # Normal WRF file
+        cen_lon = ds.CEN_LON
+        cen_lat = ds.CEN_LAT
+        dx = ds.DX
+        dy = ds.DY
+        pargs['lat_1'] = ds.TRUELAT1
+        pargs['lat_2'] = ds.TRUELAT2
```

Comment:
```suggestion
        pargs['lat_2'] = ds.TRUELAT2 if hasattr(ds, 'TRUELAT2') else ds.TRUELAT1
```

TRUELAT2 might be missing when TRUELAT1 == TRUELAT2